pkg/network/tracer/connection/tracer.go

```diff
@@ -433,34 +445,44 @@ func (t *tracer) getEBPFTelemetry() *netebpf.Telemetry {
 	return telemetry
 }
 
-func (t *tracer) RefreshProbeTelemetry() {
-	ticker := time.NewTicker(10 * time.Second)
-	defer ticker.Stop()
-	for {
-		select {
-		case <-ticker.C:
-			t.refreshProbeTelemetry()
-		case <-t.exitTelemetry:
-			return
-		}
-	}
+// Describe returns all descriptions of the collector.
+func (t *tracer) Describe(ch chan<- *prometheus.Desc) {
+	ch <- ConnTracerTelemetry.UdpDroppedConns
 }
 
-// Refreshes connection tracer telemetry on a loop
-// TODO: Replace with prometheus collector interface
-func (t *tracer) refreshProbeTelemetry() {
+// Collect returns the current state of all metrics of the collector.
+func (t *tracer) Collect(ch chan<- prometheus.Metric) {
 	ebpfTelemetry := t.getEBPFTelemetry()
 	if ebpfTelemetry == nil {
 		return
 	}
+	delta := int64(ebpfTelemetry.Tcp_failed_connect) - ConnTracerTelemetry.lastTcpFailedConnects.Load()
+	ConnTracerTelemetry.lastTcpFailedConnects.Store(int64(ebpfTelemetry.Tcp_failed_connect))
+	ch <- prometheus.MustNewConstMetric(ConnTracerTelemetry.tcpFailedConnects, prometheus.CounterValue, float64(delta))
```

Comment:
Should we use the delta or the current value of the counter here? i.e. should we even have a delta at all?

Reply:
I think we discussed this at some point, but IIRC if we don't do the delta then we have to use `monotonic_diff` when graphing the counter, so it's just a tradeoff